app/models/validators/beyond_osra_establishment_validator.rb

```diff
@@ -0,0 +1,13 @@
+class BeyondOsraEstablishmentValidator < ActiveModel::EachValidator
+  OSRA_ESTABLISHMENT_DATE = Date.new(2012,04,01)
+  include Helpers
+
+  def validate_each(record, attribute, value)
+    if !valid_date? value
+      record.errors[attribute] << ('is not a valid date')
```

Comment:
Date format will be automatically validated since the value is stored as a `date` in the db.